### PR TITLE
String function Support library

### PR DESCRIPTION
**Pull request by lsubash**
Add string functionality in Osloader to a
common library

Signed-off-by: Subash Lakkimsetti <subash.lakkimsetti@intel.com>

### CODE REVIEW

File: BootloaderCommonPkg/Library/StringSupportLib/StringSupportLib.c [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+
+  @retval      The pointer for next line.
+               NULL if no more line is available.
+**/
+
+#include <Library/StringSupportLib.h>
+
+CHAR8 *
+EFIAPI
+GetNextLine (
```
Comment (mauricema) [inline BootloaderCommonPkg/Library/StringSupportLib/StringSupportLib.c:23]: Add argument comments

File: BootloaderCommonPkg/Library/StringSupportLib/StringSupportLib.c [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+  IN   CHAR8    *Start,
+  OUT  UINT32   *Length
+  )
+{
+  CHAR8 *Ptr;
+
+  Ptr = Start;
```
Comment (mauricema) [inline BootloaderCommonPkg/Library/StringSupportLib/StringSupportLib.c:30]: Check Start is NULL,  if so, return NULL.

File: BootloaderCommonPkg/Library/StringSupportLib/StringSupportLib.c [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+  while ((Ptr[0] != 0) && (Ptr[0] != '\n')) {
+    Ptr++;
+  }
+
+  *Length = (UINT32)(Ptr - Start);
```
Comment (mauricema) [inline BootloaderCommonPkg/Library/StringSupportLib/StringSupportLib.c:35]: Only do assignment if Length != NULL

File: BootloaderCommonPkg/Library/StringSupportLib/StringSupportLib.c [hunk 4]
```diff
@@ -0,0 +1,169 @@
...
+
+  @retval      The pointer to the trimmed line.
+
+**/
+CHAR8 *
+EFIAPI
+TrimLeft (
+  IN  CHAR8  *Line
+  )
+{
```
Comment (mauricema) [inline BootloaderCommonPkg/Library/StringSupportLib/StringSupportLib.c:57]: Check Line is NULL or not

File: BootloaderCommonPkg/Library/StringSupportLib/StringSupportLib.c [hunk 5]
```diff
@@ -0,0 +1,169 @@
...
+CHAR8 *
+EFIAPI
+GetNextSpace (
+  IN  CHAR8  *LineStart,
+  IN  CHAR8  *LineEnd
+  )
+{
+  CHAR8  *Line;
+
+  Line = LineStart;
```
Comment (mauricema) [inline BootloaderCommonPkg/Library/StringSupportLib/StringSupportLib.c:83]: Check LineStart/LineEnd is NULL

File: BootloaderCommonPkg/Library/StringSupportLib/StringSupportLib.c [hunk 6]
```diff
@@ -0,0 +1,169 @@
...
+
+**/
+UINT32
+EFIAPI
+MatchAssignment (
+  IN  CHAR8  *Line,
+  IN  CHAR8  *Variable
+  )
+{
+  UINT32 VarLen;
```
Comment (mauricema) [inline BootloaderCommonPkg/Library/StringSupportLib/StringSupportLib.c:159]: Check Line/Varaible is NULL

File: BootloaderCommonPkg/Library/StringSupportLib/StringSupportLib.c [hunk 7]
```diff
@@ -0,0 +1,169 @@
...
+  @retval      Length of the matched keyword.
+               0 if there is no match.
+
+**/
+UINT32
+EFIAPI
+MatchKeyWord (
+  IN  CHAR8  *Line,
+  IN  CHAR8  *Keyword
+  )
```
Comment (mauricema) [inline BootloaderCommonPkg/Library/StringSupportLib/StringSupportLib.c:127]: Check Line/Keywork is NULL

File: BootloaderCommonPkg/Include/Library/StringSupportLib.h [hunk 8]
```diff
@@ -0,0 +1,109 @@
+/** @file
+Header file for strig function support library
```
Comment (myguitar) [inline BootloaderCommonPkg/Include/Library/StringSupportLib.h:2]: typo. strig -> string

File: BootloaderCommonPkg/Library/StringSupportLib/StringSupportLib.inf [hunk 9]
```diff
@@ -0,0 +1,27 @@
+## @file
+#  String SUpport Library Instance.
```
Comment (myguitar) [inline BootloaderCommonPkg/Library/StringSupportLib/StringSupportLib.inf:2]: typo. SUpport -> Support

File: BootloaderCorePkg/BootloaderCorePkg.dsc [hunk 10]
```diff
@@ -374,6 +374,7 @@
       AbSupportLib        | PayloadPkg/Library/AbSupportLib/AbSupportLib.inf
       SblParameterLib     | PayloadPkg/Library/SblParameterLib/SblParameterLib.inf
       TrustyBootLib       | PayloadPkg/Library/TrustyBootLib/TrustyBootLib.inf
+      StringSupportLib    | BootloaderCommonPkg/Library/StringSupportLib/StringSupportLib.inf
```
Comment (myguitar) [inline BootloaderCorePkg/BootloaderCorePkg.dsc:377]: I think we can put StringSupportLib in LibraryClasses for common use.

File: BootloaderCorePkg/BootloaderCorePkg.dsc [hunk 11]
```diff
@@ -386,6 +387,7 @@
       PayloadSupportLib       | PayloadPkg/Library/PayloadSupportLib/PayloadSupportLib.inf
       BootloaderLib           | PayloadPkg/Library/PayloadLib/PayloadLib.inf
       FirmwareUpdateLib       | Silicon/$(SILICON_PKG_NAME)/Library/FirmwareUpdateLib/FirmwareUpdateLib.inf
+      StringSupportLib        | BootloaderCommonPkg/Library/StringSupportLib/StringSupportLib.inf
```
Comment (myguitar) [inline BootloaderCorePkg/BootloaderCorePkg.dsc:390]: I think we can put StringSupportLib in LibraryClasses for common use.

**Comment by lsubash** [comment]
Thanks Maurice and Aiden. I would incorporate your feedback.

**Pull request merged**